gio/src/cancellable.rs

```diff
@@ -1,10 +1,238 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
 use crate::prelude::*;
+use crate::Cancellable;
+use futures_channel::oneshot;
+use futures_core::Future;
+use glib::{translate::*, IsA, Source};
+use std::num::NonZeroU64;
 
-#[test]
-fn check_callback() {
-    let c = crate::Cancellable::new();
-    c.connect_cancelled(|_| {});
-    c.cancel(); // if it doesn't crash at this point, then we're good to go!
+impl Cancellable {
+    // rustdoc-stripper-ignore-next
+    /// Creates a source that will never trigger.
+    pub fn pending_source() -> Source {
+        unsafe { from_glib_full(ffi::g_cancellable_source_new(std::ptr::null_mut())) }
+    }
+}
+
+// rustdoc-stripper-ignore-next
+/// The id of a cancelled handler that is returned by `CancellableExtManual::connect`. This type is
+/// analogous to [`glib::SignalHandlerId`].
+#[derive(Debug, Eq, PartialEq)]
+#[repr(transparent)]
+pub struct CancelledHandlerId(NonZeroU64);
+
+impl CancelledHandlerId {
+    // rustdoc-stripper-ignore-next
+    /// Returns the internal signal handler ID.
+    pub unsafe fn as_raw(&self) -> libc::c_ulong {
```

Comment:
That should just be `IntoGlib` or not?

Reply:
Not sure, I just copied it from `SignalHandlerId`

Reply:
I guess let's just keep that then